ui/tests/acceptance/secrets/backend/azure/azure-configuration-test.js

```diff
@@ -31,113 +33,427 @@ module('Acceptance | Azure | configuration', function (hooks) {
-      // cleanup
-      await runCmd(`delete sys/mounts/${path}`);
+    module('details', function () {
+      test('it should show configuration with Azure account options configured', async function (assert) {
+        const path = `azure-${this.uid}`;
+        const azureAccountAttrs = {
+          client_secret: 'client-secret',
+          subscription_id: 'subscription-id',
+          tenant_id: 'tenant-id',
+          client_id: 'client-id',
+          root_password_ttl: '20 days 20 hours',
+          environment: 'AZUREPUBLICCLOUD',
+        };
+        this.server.get(`${path}/config`, () => {
+          assert.ok(true, 'request made to config when navigating to the configuration page.');
+          return { data: { id: path, type: this.type, ...azureAccountAttrs } };
+        });
+        await enablePage.enable(this.type, path);
+        for (const key of expectedConfigKeys('azure')) {
+          assert.dom(GENERAL.infoRowLabel(key)).exists(`${key} on the ${this.type} config details exists.`);
+          const responseKeyAndValue = expectedValueOfConfigKeys(this.type, key);
+          assert
+            .dom(GENERAL.infoRowValue(key))
+            .hasText(responseKeyAndValue, `value for ${key} on the ${this.type} config details exists.`);
+        }
+        // check mount configuration details are present and accurate.
+        await click(SES.configurationToggle);
+        assert
+          .dom(GENERAL.infoRowValue('Path'))
+          .hasText(`${path}/`, 'mount path is displayed in the configuration details');
+        // cleanup
+        await runCmd(`delete sys/mounts/${path}`);
+      });
+
+      test('it should show API error when configuration read fails', async function (assert) {
+        assert.expect(1);
+        const path = `azure-${this.uid}`;
+        // interrupt get and return API error
+        this.server.get(configUrl(this.type, path), () => {
+          return overrideResponse(400, { errors: ['bad request'] });
+        });
+        await enablePage.enable(this.type, path);
+        assert.dom(SES.error.title).hasText('Error', 'shows the secrets backend error route');
+      });
     });
 
-    test('it should not show "Configure" from toolbar', async function (assert) {
-      const path = `azure-${this.uid}`;
-      await enablePage.enable('azure', path);
-      assert.dom(SES.configure).doesNotExist('Configure button does not exist.');
-      // cleanup
-      await runCmd(`delete sys/mounts/${path}`);
+    module('create', function () {
+      test('it should save azure account accessType options', async function (assert) {
+        assert.expect(3);
+        const path = `azure-${this.uid}`;
+        await enablePage.enable(this.type, path);
+
+        this.server.post('/identity/oidc/config', () => {
+          assert.notOk(
+            true,
+            'post request was made to issuer endpoint when on community and data not changed. test should fail.'
+          );
```

Comment:
Noelle had a great idea that instead of using `assert` here we should throw an error, like this:
https://github.com/hashicorp/vault/blob/main/ui/tests/integration/components/dashboard/overview-test.js#L105-L108

So instead of the assertion count being the only thing that lets us know this errors, we actually get an error

Reply:
I did this a couple of [places](https://github.com/hashicorp/vault/pull/29047/files#diff-0c07ee0662dabed80beda3785c427a67da0318ec4cfb75d23f8a4b2c97459541R305), but this test will fail if the server.post gets hit because [notOk](https://qunitjs.com/api/assert/notOk/) first param should be false, but here it's true (thus, the assert.expect isn't the only catch).

To me, I prefer the assertion failing like this, but I'm clearly not beholding to one way or the other having used both approaches.